process-engine-plugin/handling-jpa-variables/README.md

```diff
@@ -0,0 +1,145 @@
+# JPA serializer 
+
+This example demonstrates how to configure a JPA serializer as a Process Engine plugin.
+
+
+## Show me the important parts!
+
+
+
+### Create a Process Engine Plugin Implementation
+
+Extend the `org.camunda.bpm.engine.impl.cfg.AbstractProcessEnginePlugin` class:
+
+``` java
+public class JPAVariablesSerializerPlugin extends AbstractProcessEnginePlugin {
+
+  @Override
+  public void postInit(ProcessEngineConfigurationImpl processEngineConfiguration) {
+    EntityManagerFactory jpaEntityManagerFactory = Persistence.createEntityManagerFactory("activiti-jpa-pu");
+    processEngineConfiguration.getSessionFactories().put(EntityManagerSession.class,
+        new EntityManagerSessionFactory(jpaEntityManagerFactory, true, true));
+
+    VariableSerializers variableSerializers = processEngineConfiguration.getVariableSerializers();
+    int index = variableSerializers.getSerializerIndexByName(ValueType.BYTES.getName());
+    if (index > -1) {
+      variableSerializers.addSerializer(new JPAVariableSerializer(), index);
+    } else {
+      variableSerializers.addSerializer(new JPAVariableSerializer());
+    }
+  }
+}
+```
+
+The `persistenceUnitName` passed to the `EntityManagerFactory` is defined in a [`persistance.xml`][1] file following the JPA standarts.
+Add a `sessionFactory` with key `EntityManagerSession.class` and retrieve already initilized `variableSerializers`.
+In case the `ValueType.BYTES` replace it with the custom `JPAVariableSerializer`, otherwise just add it.
```

Comment:
```suggestion
If there is a serializer for `ValueType.BYTES` values, replace it with the custom `JPAVariableSerializer`. Otherwise, add the JPA serializer to the list.
```